src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -34,11 +33,24 @@ internal static SafeSslHandle AllocateSslContext(long options, SafeX509Handle ce
 
                 libssl.SSL_CTX_set_quiet_shutdown(innerContext, 1);
 
+                libssl.SSL_CTX_set_cipher_list(innerContext, encryptionPolicy);
+
                 if (certHandle != null && certKeyHandle != null)
                 {
                     SetSslCertificate(innerContext, certHandle, certKeyHandle);
                 }
 
+                if (remoteCertRequired)
+                {
+                    Debug.Assert(isServer, "isServer flag should be true");
+                    libssl.SSL_CTX_set_verify(innerContext,
+                    (int)libssl.ClientCertOption.SSL_VERIFY_PEER |
+                    (int)libssl.ClientCertOption.SSL_VERIFY_FAIL_IF_NO_PEER_CERT, verify_callback);
```

Comment:
I'm not 100% sure about the delegate object lifetime, but I think you want to save the delegate to a field; otherwise the GC can purge it at an arbitrary point and then your callback will be to an invalid function pointer.